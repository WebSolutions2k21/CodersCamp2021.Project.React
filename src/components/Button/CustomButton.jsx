import { ThemeProvider } from '@mui/material/styles';
import Button from '@mui/material/Button';

import { ButtonTheme } from '../../styles/themes/CustomButtonTheme';

export const CustomButton = ({ text, color, size, clickAction, disabled }) => {
  const handleClick = () => {
    clickAction && clickAction();
  };

  return (
    <ThemeProvider theme={ButtonTheme}>
      {clickAction ? (
        <Button variant="contained" color={color} size={size} onClick={handleClick} disabled={disabled}>
          {text}
        </Button>
      ) : (
        <Button type="submit" variant="contained" color={color} size={size} disabled={disabled}>
          {text}
        </Button>
      )}
    </ThemeProvider>
  );
};
