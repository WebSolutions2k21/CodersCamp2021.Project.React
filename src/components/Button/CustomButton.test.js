import { render, screen, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { CustomButton } from './CustomButton';


afterEach(() => {
  cleanup();
});

test('1. should render CustomButton component with onClick function', () => {
  render(<CustomButton text="Add Pet" color="primary" size="small" clickAction={()=> console.log('CustomButton')}/>);
  const inputComponent = screen.getByRole('button', {name: /Add Pet/i})
  expect(inputComponent).toBeInTheDocument();
  expect(inputComponent).not.toBeNull();
  expect(inputComponent).not.toBeDisabled();
  expect(inputComponent).toHaveTextContent('Add Pet');
});

test('2. should render CustomButton component without onClick function', () => {
  render(<CustomButton text="Add Pet" color="primary" size="small"/>);
  const inputComponent = screen.getByRole('button', {name: /Add Pet/i})
  expect(inputComponent).toBeInTheDocument();
  expect(inputComponent).not.toBeNull();
  expect(inputComponent).not.toBeDisabled();
  expect(inputComponent).toHaveTextContent('Add Pet');
});

test('3. should render disabled CustomButton', () => {
    render(<CustomButton text="Add Pet" color="primary" size="small" disabled/>);
    const inputComponent = screen.getByRole('button', {name: /Add Pet/i})
    expect(inputComponent).toBeDisabled();
});

test('4. should fire CustomButton function when clicked', () => {
    const handleClick = jest.fn()
    render(<CustomButton text="Add Pet" color="primary" size="small" clickAction={() => handleClick()}/>);
    const inputComponent = screen.getByRole('button', {name: /Add Pet/i})
    fireEvent.click(inputComponent);
    expect(handleClick).toHaveBeenCalledTimes(1)
});

test('matches snapshot', () => {
  const tree = renderer.create(<CustomButton text="Add Pet" color="primary" size="small" />).toJSON();
  expect(tree).toMatchSnapshot();
});