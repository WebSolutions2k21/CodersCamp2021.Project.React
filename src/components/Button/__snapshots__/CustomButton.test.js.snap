// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButtonBase-root  css-1arvnqi-MuiButtonBase-root-MuiButton-root"
  disabled={false}
  onBlur={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="submit"
>
  Add Pet
</button>
`;
